### ISSUE TITLE
getstorageinfo:: AttributeError: 'dict' object has no attribute 'split'

### ISSUE THREAD

**Issue by quickdraw6906** [bkerler/edl#362]
Partial output including error:
```
-----------------
program,read,nop,patch,configure,setbootablestoragedrive,erase,power,firmwarewrite,getstorageinfo,benchmark,emmc,ufs,fixgpt,getsha256digest,gethwversion,getrfversion,getprjversion,setprojmodel,demacia,sha256init,sha256final,eraseuserdata
oneplus - Oneplus protection with prjid 18821 detected
firehose - GetStorageInfo:
--------------------

Traceback (most recent call last):
  File "/Users/sean/OnePlus 7 Pro/root/edl/./edl", line 391, in <module>
    base.run()
  File "/Users/sean/OnePlus 7 Pro/root/edl/./edl", line 386, in run
    fh.handle_firehose(cmd, options)
  File "/Users/sean/OnePlus 7 Pro/root/edl/edlclient/Library/firehose_client.py", line 647, in handle_firehose
    return self.firehose.cmd_getstorageinfo_string()
  File "/Users/sean/OnePlus 7 Pro/root/edl/edlclient/Library/firehose.py", line 1308, in cmd_getstorageinfo_string
    data = self.xml.getlog(val.data)
  File "/Users/sean/OnePlus 7 Pro/root/edl/edlclient/Library/xmlparser.py", line 29, in getlog
    lines = input.split(b"<?xml")
AttributeError: 'dict' object has no attribute 'split'
```
The issue is getstorageinfo is returnin JSON:
```
{'value': 'ACK', 'rawmode': 'false'}
```
Changing getlog() to this:
```
  def getlog(self, input):
        data = []
        try:
          lines = input.split(b"<?xml")
        except Exception:
            data.append(input)
            return data
   snip...
```
Output now:
```
firehose_client - Supported functions:
-----------------
program,read,nop,patch,configure,setbootablestoragedrive,erase,power,firmwarewrite,getstorageinfo,benchmark,emmc,ufs,fixgpt,getsha256digest,gethwversion,getrfversion,getprjversion,setprojmodel,demacia,sha256init,sha256final,eraseuserdata
oneplus - Oneplus protection with prjid 18821 detected
firehose - GetStorageInfo:
--------------------

firehose - {'value': 'ACK', 'rawmode': 'false'}
```

**Comment by mans0n** [comment]
Had the same error. My hacky patch:

```diff
--- a/edlclient/Library/firehose.py
+++ b/edlclient/Library/firehose.py
@@ -281,7 +281,7 @@ class firehose(metaclass=LogBase):
                         if resp["rawmode"] == "false":
                             if status:
                                 log = self.xml.getlog(rdata)
-                                return response(resp=status, data=resp, log=log)
+                                return response(resp=status, data=rdata, log=log)
                             else:
                                 error = self.xml.getlog(rdata)
                                 return response(resp=status, error=error, data=resp, log=error)
```

Don't know whether the other lines also need some love.

**Comment by bkerler** [comment]
Fixed in commit e5ec2ca. Thanks for pointing that out.
